vpcs_test.go

```diff
@@ -176,46 +178,76 @@ func TestVPCs_Update(t *testing.T) {
 }
 
 func TestVPCs_Set(t *testing.T) {
-	setup()
-	defer teardown()
 
-	type setRequest struct {
-		Name        string `json:"name"`
-		Description string `json:"description"`
-	}
+	tests := []struct {
+		desc                string
+		id                  string
+		updateFields        []VPCSetField
+		mockResponse        string
+		expectedRequestBody string
+		expectedUpdatedVPC  *VPC
+	}{
+		{
+			desc: "setting name and description",
+			id:   "880b7f98-f062-404d-b33c-458d545696f6",
+			updateFields: []VPCSetField{
+				VPCSetName("my-new-vpc"),
+				VPCSetDescription("vpc description"),
+			},
+			mockResponse: `
+			{
+			  "vpc":
+			` + vTestJSON + `
+			}
+			`,
+			expectedRequestBody: `{"description":"vpc description","name":"my-new-vpc"}`,
+			expectedUpdatedVPC:  vTestObj,
+		},
 
-	svc := client.VPCs
-	path := "/v2/vpcs"
-	want := vTestObj
-	id := "880b7f98-f062-404d-b33c-458d545696f6"
-	name := "my-new-vpc"
-	desc := "vpc description"
-	req := &setRequest{
-		Name:        name,
-		Description: desc,
+		{
+			desc: "setting the default vpc option",
+			id:   "880b7f98-f062-404d-b33c-458d545696f6",
+			updateFields: []VPCSetField{
+				VPCSetName("my-new-vpc"),
+				VPCSetDescription("vpc description"),
+				VPCSetDefault(),
+			},
+			mockResponse: `
+			{
+			  "vpc":
+			` + vTestJSON + `
+			}
+			`,
+			expectedRequestBody: `{"default":true,"description":"vpc description","name":"my-new-vpc"}`,
+			expectedUpdatedVPC:  vTestObj,
+		},
 	}
-	jsonBlob := `
-{
-  "vpc":
-` + vTestJSON + `
-}
-`
 
-	mux.HandleFunc(path+"/"+id, func(w http.ResponseWriter, r *http.Request) {
-		c := new(setRequest)
-		err := json.NewDecoder(r.Body).Decode(c)
-		if err != nil {
-			t.Fatal(err)
-		}
+	for _, tt := range tests {
+		setup()
 
-		testMethod(t, r, http.MethodPatch)
-		require.Equal(t, c, req)
-		fmt.Fprint(w, jsonBlob)
-	})
+		mux.HandleFunc("/v2/vpcs/"+tt.id, func(w http.ResponseWriter, r *http.Request) {
+			buf := new(bytes.Buffer)
+			buf.ReadFrom(r.Body)
+			require.Equal(t, tt.expectedRequestBody, strings.TrimSpace(buf.String()))
 
-	got, _, err := svc.Set(ctx, id, VPCSetName(name), VPCSetDescription(desc))
-	require.NoError(t, err)
-	require.Equal(t, want, got)
+			v := new(VPCUpdateRequest)
+			err := json.NewDecoder(buf).Decode(v)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			testMethod(t, r, http.MethodPatch)
```

Comment:
Would be nice to also cover default VPC from `Update` test case at line #L144

Reply:
Good catch! Missed this.